service/controller/v24/resource/cpi/create.go

```diff
@@ -49,25 +49,19 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "computing the template of the tenant cluster's control plane initializer CF stack")
 
-		cc, err := controllercontext.FromContext(ctx)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-
-		var cpi *adapter.CPI
+		var params *template.ParamsMain
```

Comment:
I would drop `Main` suffix. Not a blocker.

Reply:
The thing is that we have params for each template and that reflects the structure well, even though it is not the most pretty solution.